lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -642,6 +642,14 @@ def _on_update_status(self, event: UpdateStatusEvent):  # noqa: C901
         ):
             self.opensearch_provider.remove_lingering_relation_users_and_roles()
 
+        # If the unit reloads its certs but the other units are not ready yet
+        # we need to wait for them all to be ready before deleting the old CA
+        if (
+            self.tls.read_stored_ca("old-ca")
+            and self.tls.ca_and_certs_rotation_complete_in_cluster()
+        ):
+            logger.debug("update_status: Detected CA rotation complete in cluster")
+            self.tls.on_ca_certs_rotation_complete()
```

Comment:
I've specifically tested this manually on my local machine, and unfortunately it failed.

This is the setup:
- deploy 3 nodes of Opensearch, run the usual CA rotation via triggering a config change
- waiting for the CA rotation to complete, then remove two units and only keep one of those that did not finish last on the first CA rotation (to enforce the `old-ca` to still be there)
- wait for one (or several) `update-status` to pass
- trigger another CA rotation on that one node

This fails then with `Destination alias <old-ca> already exists` on `CertificateAvailableEvent`. That means the `old-ca` was not removed on the `update_status` runs. I am still trying to figure out why, didn't find the root cause yet.

Reply:
After rotating the CA and applying the new certs, the `old-ca` is now removed correctly.